framework/src/st_things/things_stack/src/common/framework/things_security_manager.c

```diff
@@ -226,10 +226,32 @@ static OCStackResult seckey_setup(const char *filename, OicSecKey_t *key, OicEnc
 		return OC_STACK_ERROR;
 	}
 
+	if (encoding == OIC_ENCODING_UNKNOW) {
+		char *file_ext;
+		int filename_len = strlen(filename);
+		file_ext = filename + filename_len;
+
+		while(file_ext >= filename) {
+			if (*file_ext == '.') {
+				file_ext += 1;
+				break;
+			}
+			file_ext--;
+		}
+
+		if (strncmp(file_ext, "PEM", 3) == 0 || strncmp(file_ext, "pem", 3) == 0) {
```

Comment:
If there is no '.', file_ext pointer will be (filename - 1). At that case, is it right to compare with "PEM"?